src/Managers/SchemaManager.php

```diff
@@ -48,10 +55,90 @@ public function drop(string $documentClass): bool
         $document = $this->manager->getDocumentObject($documentClass);
         $tableName = $this->manager->getTableName($document->getTableName());
 
-        $this->manager->getDbClient()->deleteTable([
-            self::TABLE_NAME_KEY => $tableName
-        ]);
+        try {
+            $this->manager->getDbClient()->deleteTable([
+                self::TABLE_NAME_KEY => $tableName
+            ]);
+        } catch (DynamoDbException $exception) {
+            if ($this->canHandleDropException($exception) === false) {
+                throw $exception;
+            }
+        }
 
         return true;
     }
+
+    /**
+     * Assert if an exception while creating can be handled and
+     * operation can be continued.
+     *
+     * @param \Aws\DynamoDb\Exception\DynamoDbException $exception
+     *
+     * @return bool
+     */
+    private function canHandleCreateException(DynamoDbException $exception): bool
+    {
+        /**
+         * AWS throws ResourceInUseException when you try to create an existing table.
+         *
+         * @see https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_CreateTable.html#API_CreateTable_Errors
+         */
+        return $this->canHandleException(
+            'CreateTable',
+            ['ResourceInUseException'],
+            $exception,
+            400
+        );
+    }
+
+    /**
+     * Assert if an exception while dropping can be handled and
+     * operation can be continued.
+     *
+     * @param \Aws\DynamoDb\Exception\DynamoDbException $exception
+     *
+     * @return bool
+     */
+    private function canHandleDropException(DynamoDbException $exception): bool
+    {
+        /**
+         * AWS can throw ResourceInUseException or ResourceNotFoundException
+         * when you try to delete a table currently in use or non existent table.
+         *
+         * @see https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_DeleteTable.html#API_DeleteTable_Errors
+         */
+        return $this->canHandleException(
+            'DeleteTable',
+            ['ResourceInUseException', 'ResourceNotFoundException'],
```

Comment:
yes in case the table is being read by something else. this is based on the docs. I am not sure if we want to be able to handle that scenario. 
https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_DeleteTable.html#API_DeleteTable_Errors

Reply:
fixed.